src/components/transport_manager/test/transport_manager_default_test.cc

```diff
@@ -110,80 +134,178 @@ TEST(TestTransportManagerDefault, Init_LastStateNotUsed) {
       .WillRepeatedly(ReturnRef(dummy_parameter));
-  transport_manager.Init(mock_last_state);
-  transport_manager.Stop();
-}
 
-TEST(TestTransportManagerDefault, Init_LastStateUsed) {
-  MockTransportManagerSettings transport_manager_settings;
-  transport_manager::TransportManagerDefault transport_manager(
-      transport_manager_settings);
+  // Arrange necessary transport adapters mock objects
+  MockBluetoothTransportAdapter* mock_bt_ta =
+      new MockBluetoothTransportAdapter();
 
-  NiceMock<MockLastState> mock_last_state;
-  Json::Value custom_dictionary;
-  Json::Value tcp_device;
-  tcp_device[kDeviceName] = "unique_tcp_device_name";
-  tcp_device[kDeviceAddress] = "127.0.0.1";
-  tcp_device[kDeviceApplications][0][kApplicationPort] = kApplicationPortValue;
-  Json::Value bluetooth_device;
-  bluetooth_device[kDeviceName] = "unique_bluetooth_device_name";
-  bluetooth_device[kDeviceAddress] = "AB:CD:EF:GH:IJ:KL";
-  bluetooth_device[kDeviceApplications][0][kApplicationRfcomm] =
-      kApplicationRfcommValue;
-  custom_dictionary[kTransportManager][kTcpAdapter][kDevices][0] = tcp_device;
-  custom_dictionary[kTransportManager][kBluetoothAdapter][kDevices][0] =
-      bluetooth_device;
+  MockTCPTransportAdapter* mock_tcp_ta = new MockTCPTransportAdapter(
+      tcp_adapter_port, mock_last_state, transport_manager_settings);
 
-  ON_CALL(mock_last_state, get_dictionary())
-      .WillByDefault(ReturnRef(custom_dictionary));
+  MockUsbAoaAdapter* mock_usb_aoa_ta =
+      new MockUsbAoaAdapter(mock_last_state, transport_manager_settings);
 
-  EXPECT_CALL(transport_manager_settings, use_last_state())
+  MockCloudWebsocketTransportAdapter* mock_cloud_websocket_ta =
+      new MockCloudWebsocketTransportAdapter(mock_last_state,
+                                             transport_manager_settings);
+
+  // Expectations for Mock of bluetooth transport adapter
+  EXPECT_CALL(*mock_bt_ta, AddListener(_));
+  EXPECT_CALL(*mock_bt_ta, IsInitialised()).WillRepeatedly(Return(true));
+  EXPECT_CALL(*mock_bt_ta, Init()).WillRepeatedly(Return(TransportAdapter::OK));
+  EXPECT_CALL(*mock_bt_ta, Terminate());
+
+  // Expectations for Mock of TCP transport adapter
+  EXPECT_CALL(*mock_tcp_ta, AddListener(_));
+  EXPECT_CALL(*mock_tcp_ta, IsInitialised()).WillRepeatedly(Return(true));
+  EXPECT_CALL(*mock_tcp_ta, Init())
+      .WillRepeatedly(Return(TransportAdapter::OK));
+
+  std::shared_ptr<MockDevice> mockdev =
+      std::make_shared<MockDevice>(dev_id, unique_tcp_dev_name);
+  EXPECT_CALL(*mock_tcp_ta, FindDevice(unique_tcp_dev_name))
+      .WillRepeatedly(Return(mockdev));
+  const int app_handle = 1;
+  EXPECT_CALL(*mock_tcp_ta, Connect(unique_tcp_dev_name, app_handle)).Times(0);
+  EXPECT_CALL(*mock_tcp_ta, Terminate());
+
+  // Expectations for Mock of USB transport adapter
+  EXPECT_CALL(*mock_usb_aoa_ta, AddListener(_));
+  EXPECT_CALL(*mock_usb_aoa_ta, IsInitialised()).WillRepeatedly(Return(true));
```

Comment:
@SNiukalov Such a big test. Can't we split it? I think at least we can move precondition (device name, address etc) to the class method and call it from test.
Also we can move to methods: Expectations for Mock of bluetooth transport adapter
and Expectations for Mock of USB transport adapter
and Expectations for Mock of Cloud Websocket transport adapter
because also we can call them from other tests

Reply:
@ZhdanovP 
Fixed in:https://github.com/smartdevicelink/sdl_core/pull/3134/commits/f653a628ed63e5a95fa2564a8700c348b865b70b